pkgs/tools/typesetting/tectonic/default.nix

```
@@ -30,9 +30,17 @@ rustPlatform.buildRustPackage rec {
     sha256 = "sha256-xZHYiaQ8ASUwu0ieHIXcjRaH06SQoB6OR1y7Ok+FjAs=";
   };
 
+  patches = [
+    (fetchpatch2 {
+      name = "1155-fix-endless-reruns-when-generating-bbl";
+      url = "https://github.com/tectonic-typesetting/tectonic/commit/fbb145cd079497b8c88197276f92cb89685b4d54.patch";
+      hash = "sha256-6FW5MFkOWnqzYX8Eg5DfmLaEhVWKYVZwodE4SGXHKV0=";
+    })
+  ];
+
   cargoPatches = [
-    # fix build with rust 1.80
     (fetchpatch2 {
+      name = "1202-fix-build-with-rust-1_80";
```

I think this causes the hash to change. Could you verify that? I think you can do it with `nix build -Lf. tectonic.cargoPatches --rebuild`. I personally prefer comments over, especially if the comments include a link, which is way more useful then a number that suggests it is the number of the PR...

---

> I think this causes the hash to change. Could you verify that? I think you can do it with `nix build -Lf. tectonic.cargoPatches --rebuild`.

Yeah, I also thought so and tested this, and it turns out that changing the `name` _will_ trigger a rebuild but will _not_ change the hash. Confirmed again by running `--rebuild`.

> I personally prefer comments over, especially if the comments include a link, which is way more useful then a number that suggests it is the number of the PR...

Can we do both? :laughing: like this:

```suggestion
      # cherrypicked from https://github.com/tectonic-typesetting/tectonic/pull/1202
      name = "1202-fix-build-with-rust-1_80";
```

The `name` makes debugging the build a bit more human-friendly :laughing: P.S. `cherrypicked from ...` because upstream 1202 contains more commits than the one we currently want (for the rust 1.80 fix).

---

> Yeah, I also thought so and tested this, and it turns out that changing the `name` _will_ trigger a rebuild but will _not_ change the hash. Confirmed again by running `--rebuild`.

OK Great to know, thanks for checking :).

> Can we do both? 😆 like this:
> 
> The `name` makes debugging the build a bit human-friendlier 😆 P.S. `cherry-picked from ...` because upstream PR 1202 contains more commits than the one we currently want (for the rust 1.80 fix).

Definitely I agree with you. I once saw an issue that suggests to make `fetchFromGitHub` use a better `name` by default, based on the `repo` and `owner` argument, but I can't find it.